### ISSUE TITLE
Async race condition when compressing and renaming

### ISSUE THREAD

**Issue by bogdanlupuro** [smh/grunt-maven-tasks#48]
On the continue integration server I have a problem. I sometime get an error that it can't rename the zip artifact before deploying it to artifactory. I have isolated the problem to `mvn:package` task. The problem is that it starts packaging the folder but does not wait to finish before going to rename the function, so it is possible that the archive is not yet created. The rename should be called in the callback of the `compress.tar` and after that the async should be finished.

Or am I missing something?

**Comment by smh** [comment]
That sounds reasonable. The tests use small projects, so are unlikely to catch that. I would be very happy if you could provide a pull request fixing this.

**Comment by bogdanlupuro** [comment]
Yeah, on my local computer I tried with very large projects (100mb) and couldn't reproduce. But on the integration server it happens random with very small ones.
